test/finders/songs_finder_test.rb

```diff
@@ -34,6 +35,11 @@ class SongsFinderTest < ActionController::TestCase
   end
 
   test 'retrieve all returns many songs' do
-    assert(SongsFinder.retrieve_all.count > 7) 
+    assert(SongsFinder.retrieve_all.count > 7)
+  end
+
+  test 'should partially match' do
+    assert_includes(partial_word_results, songs(:hands_to_the_heaven))
+    assert_includes(partial_word_results, songs(:glorious_day))
   end
```

Comment:
Ah let me explain: make sure the search is effectively `term1 term2 partial-term3*` and not 'term1* term2* term3*'